doc/source/rllib/rllib-catalogs.rst

```diff
@@ -0,0 +1,151 @@
+After reading this user guide you will be able to…
+
+    - Instantiate and interact with a Catalog
+    - Inject your custom models into RLModules
+    - Inject your custom action distributions into RLModules
+    - Extend RLlib’s selection of Models and distributions with your own
+    - Write a Catalog from scratch
+
+Instantiate and interact with a Catalog
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following three examples show variations of basic interaction with RLLib’s Catalogs.
+We create models as well as distributions and step through an environment with them.
+
+.. tabbed:: Basic interaction
+
+    In this example, we showcase the general API for interacting with Catalogs.
+
+    .. literalinclude:: ../../../rllib/examples/catalog/basics/basic_interaction.py
+       :language: python
+       :start-after: __sphinx_doc_begin__
+       :end-before: __sphinx_doc_end__
+
+.. tabbed:: PPOCatalog + CartPole
+
+    In this example, we showcase how to use the PPOCatalog to create models and an action distribution.
+    This is more similar to what RLlib does internally.
+
+    .. literalinclude:: ../../../rllib/examples/catalog/basics/cartpole_models_ppo.py
+       :language: python
+       :start-after: __sphinx_doc_begin__
+       :end-before: __sphinx_doc_end__
+
+.. tabbed:: Catalog + Custom Model + CartPole
+
+    In this example, we showcase how to use the base Catalog to create an encoder and an action distribution.
+    Besides these, we create a head network that fits these two by hand to show how you can combine RLLib's
+    ModelConfig API and Catalog. Extending Catalog to also build this head is how Catalog is meant to be
+    extended, which we cover later in this guide.
+
+    .. literalinclude:: ../../../rllib/examples/catalog/basics/cartpole_models.py
+       :language: python
+       :start-after: __sphinx_doc_begin__
+       :end-before: __sphinx_doc_end__
+
+What are Catalogs
+~~~~~~~~~~~~~~~~~
+
+Catalogs have two primary roles: Choosing the right model and choosing the right action distribution.
+By default, all catalogs implement decision trees that choose these based on a range of inputs.
+These include mainly the observation- and action-space of the RLModule, the model config dict and the tensor backend.
+Catalogs are also similar to what was called ModelCatalog in earlier versions of RLlib, but are more flexible.
+
+.. tabbed:: Catalogs vs RLModules
+
+    The following, broken down, diagram shows the general flow of information towards models and distributions.
+    RLModules create an instance of the Catalog class they receive as part of their constructor.
+    They then create their internal models and action distributions with the help of this Catalog.
+
+    .. note::
+      You can also modify the models and distributions in RLModules directly by overriding their constructor!
+
+    .. image:: images/catalog/catalog_and_rlm_diagram.svg
+        :align: center
+
+.. tabbed:: PPOCatalog vs PPORLModule
```

Comment:
Got it.
What do you think about drop-downs?
I want to shield users from lenthy code sections that are not 100% relevant for what comes later in the doc.
Then, you can read through the doc with ease until you hit a section that you want to expand on because you have hit the depth that suits your use case.
For example here, you would not see the content of "PPOCatalog vs PPORLModule".

Reply:
I think drop down makes more sense. Also instead of PPOCatalog vs. PPORLModule you can title it as Example of catalog in PPORLModule